compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -2083,6 +2049,114 @@ static LogicalResult adjustTileSizesForUnPackOp(func::FuncOp entryPointFn,
                                                tileSizesList, pipeline);
 }
 
+/// Set the lowering configs for all the compute ops. The lowering config is
+/// already set on `rootOperation`. We will duplicate the tile sizes of
+/// distribution and common parallel dims to other compute ops (so they have
+/// consistent configuraionts); set the tile size for the rest of dims. E.g., it
+/// sets the lowering_config for reduction + broadcast + pack op like:
+///
+///   %11 = linalg.fill ... -> tensor<384xf32>
+///   %12 = linalg.generic {
+///     indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>,
+///                      affine_map<(d0, d1) -> (d0)>,
+///                      affine_map<(d0, d1) -> (d0)>],
+///     iterator_types = ["parallel", "reduction"]}
+///     ins(%5, %6 : tensor<384x528xf32>, tensor<384xf32>)
+///     outs(%11 : tensor<384xf32>) {
+///   ^bb0(%in: f32, %in_2: f32, %out: f32):
+///     ...
+///   } -> tensor<384xf32>
+///   %13 = linalg.generic {
+///     indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>,
+///                      affine_map<(d0, d1) -> (d0)>,
+///                      affine_map<(d0, d1) -> (d0, d1)>],
+///     iterator_types = ["parallel", "parallel"]}
```

Comment:
What would happen if we had more dimensions here? E.g., ["parallel", "parallel", "parallel", "parallel"]

Reply:
Say that the lowering_config of rootOp is `[[X, 0], [Y, 0], [0, Z], [0, 0]]`, it will have [[X, 0, 0, 0], [Y, 0, 0, 0], [0, 0, 0, 0], [0, A, B, C]]`.

They have shared tile sizes for distribution and TileAndFuse, so `X` and `Y` are used in the first two list. There are no reduction dims, so the third list is empty. The `[0, A, B, C]` is derived from the characteristic of elementwise op (e.g., `getMinTilingSizesForEachDim`).